File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int[] arr) {
        ArrayList<Integer> numList = new ArrayList<>();
        
        numList.add(arr[0]);
        
        for (int i = 1; i < arr.length; i++) {
            if (arr[i] != arr[i - 1]) {
                numList.add(arr[i]);
            }
        }
        
        int[] answer = new int[numList.size()];
        for (int i = 0; i < numList.size(); i++) {
            answer[i] = numList.get(i);
        }
        
        return answer;
    }
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
function solution(arr)
{
    let answer = [];
    let current = null;
    for(let i = 0; i < arr.length; i++){
        if(arr[i] != current) {
            answer.push(arr[i])
            current = arr[i];
        }
    }
    return answer;
}